src/core/ext/transport/chttp2/transport/chttp2_transport.c

```diff
@@ -543,6 +543,10 @@ static void init_transport(grpc_exec_ctx *exec_ctx, grpc_chttp2_transport *t,
         exec_ctx, &t->keepalive_ping_timer,
         gpr_time_add(gpr_now(GPR_CLOCK_MONOTONIC), t->keepalive_time),
         &t->init_keepalive_ping_locked, gpr_now(GPR_CLOCK_MONOTONIC));
+  } else {
+    /* Use GRPC_CHTTP2_KEEPALIVE_STATE_DYING to indicate there are no inflight
```

Comment:
Thanks a lot for the review. Added `GRPC_CHTTP2_KEEPALIVE_STATE_DISABLED` to indicate the keepalive timers are not set.